examples/kustomization/operator-external-idp-oid/README.md

```diff
@@ -0,0 +1,89 @@
+# Operator Console SSO with OpenID
+
+Operator Console supports authentication with a Kubernetes Service Account Json Web Token (JWT) or OpenID. This guide explains how to configure OpenID authentication for Operator Console using the [OpenID Authorization Code Flow](https://openid.net/specs/openid-connect-core-1_0.html#CodeFlowAuth).
+
+Note: only one authentication method can be enabled at the same time, either JWT or OpenID.
+
+The `kustomization.yaml` file provided in this directory installs Operator and applies the basic configurations to enable OpenID authentication for Operator Console. Modify its environment variable values as needed for your deployment and provide the CA certificate in `console-deployment.yaml` and `console-tls-secret.yaml`.
+
+```shell
+kubectl apply -k examples/kustomization/operator-external-idp-oid/
+```
+
+### IDP Server
+
+Specify the OpenID server URL in the Operator Console Deployment by setting the `CONSOLE_IDP_URL` environment variable. This value should point to the appropriate OpenID Endpoint configuration, for example: `https://your-extenal-idp.com/.well-known/openid-configuration`.
+
+Also provide the Certificate Authority (CA) that signed the certificate the IDP server presents. You can do this by mounting a secret containing the certificate `ca.crt`. For example:
+
+For a CA certificate resembling the following:
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: idp-ca-tls
+  namespace: minio-operator
+type: Opaque
+stringData:
+  ca.crt: |
+    <CA public certificate content in plain text here> 
+```
+
+Mount the secret in the Deployment as follows:
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: console
+  namespace: minio-operator
+spec:
+  template:
+    spec:
+      containers:
+        - name: console
+          volumeMounts:
+            - mountPath: /tmp/certs/CAs
+              name: idp-certificate
+      volumes:
+        - name: idp-certificate
+          projected:
+            sources:
+              - secret:
+                  items:
+                    - key: ca.crt
+                      path: idp.crt
+                  name: idp-ca-tls
+...
+```
+
+### Client credentials
+
+Operator Console is a standalone application that identifies itself to the OpenID server using *client credentials*. The client credentials are set in the Operator Console with the following environment variables: 
+- `CONSOLE_IDP_CLIENT_ID` (client id)
+- `CONSOLE_IDP_SECRET` (client secret)
+
```

Comment:
### Access Management

All users in the OIDC realm have access to the Operator Console upon successful authentication.

To restrict access, create a new OIDC realm and use the client ID/Secret for that realm when configuring OIDC.